packages/core/ReExports/modules.tsx

```diff
@@ -192,11 +192,291 @@ const Attributes = lazy(() => import('./Attributes'))
 const FeatureDetails = lazy(() => import('./FeatureDetails'))
 const BaseCard = lazy(() => import('./BaseCard'))
 
-// uses 'as any' because otherwise typescript gives warning Exported variable
-// 'libs' has or is using name 'DataGridComponent' from external module
-// "node_modules/@mui/x-data-grid/DataGrid/DataGrid" but cannot be named.
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const DataGrid = lazy(() => import('./DataGrid')) as any
+const DataGridEntries: Record<string, LazyExoticComponent<any>> = {
+  DataGrid: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({ default: module.DataGrid })),
+  ),
+  GridActionsCellItem: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridActionsCellItem,
+    })),
+  ),
+  GridAddIcon: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridAddIcon,
+    })),
+  ),
+  GridArrowDownwardIcon: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridArrowDownwardIcon,
+    })),
+  ),
+  GridArrowUpwardIcon: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridArrowUpwardIcon,
+    })),
+  ),
+  GridAutoSizer: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridAutoSizer,
+    })),
+  ),
+  GridCellCheckboxForwardRef: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridCellCheckboxForwardRef,
+    })),
+  ),
+  GridCellCheckboxRenderer: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
+      default: module.GridCellCheckboxRenderer,
+    })),
+  ),
+  GridCheckCircleIcon: lazy(() =>
+    import('@mui/x-data-grid').then(module => ({
```

Comment:
this is probably ok, but note that this would put everything from x-data-grid in one bundle. splitting into separate might get slightly more efficient splitting but the main goal is kicking x-data-grid into its own bundle, so this isn't too bad  https://reactjs.org/docs/code-splitting.html#named-exports